### PR TITLE
Huaijiangzhu/build doc

### PR DESCRIPTION
**Pull request by MaximilienNaveau**
[//]: # "Thanks for your contribution.  To make life of the reviewers easier,"
[//]: # "please give this pull request a meaningful title and provide the"
[//]: # "requested information below."

## Description

[//]: # "Description of what you did.  If it is more complex, consider to add a"
[//]: # "'Summary' section."

fix the issue of compatibility between pip and colcon adding the mpi_cmake_modules as `test_dependencies`
build doc and refactor the setup.py a little bit.

## How I Tested

[//]: # "Explain how you tested your changes"

run `pip install .` and `colcon build`

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [x] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [x] All new functions/classes are documented and existing documentation is updated according to changes.
- [x] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

**Review by Lhumd — APPROVED** [review]

**Comment by MaximilienNaveau** [comment]
@huaijiangzhu and @jviereck I would like to make sure you saw the PR before merging.

File: readme.md [hunk 1]
```diff
@@ -1,38 +1,86 @@
...
+cd ~/devel/workspace/src
+git clone git@github.com:machines-in-motion/mpi_cmake_modules.git
+```
+or
+```
+treep --clone mpi_cmake_modules
+```
 
-Copyright(c) 2018-2021 Max Planck Gesellschaft, New York University
+TODO: *Where to find the last built doc on the internet.*
```
Comment (jviereck) [inline readme.md:79]: Do you want to keep this TODO in the official readme?

**Review by jviereck — APPROVED** [review]

File: readme.md (hunk 1 above)
Comment (MaximilienNaveau) [reply readme.md:79]: yes because I want to remember to put a link here

Comment (MaximilienNaveau) [reply readme.md:79]: we can comment this out but I don't know how to...

Comment (jviereck) [reply readme.md:79]: I don't think there is a comment feature for Markdown. So you would have to remove it.

**Pull request merged**